### ISSUE TITLE
FR: ACLs for generating HTTPS certificates

### ISSUE THREAD

**Issue by dcarrion87** [tailscale/tailscale#5917]
### What are you trying to do?

- Restrict machines that can get HTTPS certificate via Tailscale HTTPS feature.

### How should we solve this?

- Provide ACL feature to restrict which machines / labels can request certificate via Tailscale HTTPS feature.

### What is the impact of not solving this?

- We don't want general users generating certificates.

**Comment by mayakacz** [comment]
> We don't want general users generating certificates.

What's the harm if they do?

**Comment by bradfitz** [comment]
> > We don't want general users generating certificates.
>
> What's the harm if they do?

That was was my first thought too. I suppose it does mean that the host name goes on the public CT ledger.

The admin who agreed to the CT ledger warnings might not be the same person who's just playing around with the CLI or Caddy or something using the local API and making certs without knowing what that implies.

So seems like a reasonable request.

**Comment by dcarrion87** [comment]
Yep we want to control machine names that can appear on the public ledger.

**Comment by bradfitz** [comment]
Marking needs-decision so we can discuss various ACL options.

**Comment by bradfitz** [comment]
@maisem @mayakacz, nodeAttrs?

**Comment by DentonGentry** [comment]
Using node attributes for this is being discussed internally.

**Comment by michaellee8** [comment]
Is this feature being worked on? Would love to be able to avoid accidentally publishing hostnames as well.
